lib/order.rb

```diff
@@ -0,0 +1,106 @@
+require 'csv'
+require_relative 'customer'
+
+class Order
+  VALID_STATUS = [:pending, :paid, :processing, :shipped, :complete]
+  
+  attr_reader :id
+  attr_accessor :products, :customer, :fulfillment_status
+  
+  def initialize(id, products, customer, fulfillment_status = :pending)
+    @id = id
+    @products = products
+    @customer = customer
+    @fulfillment_status = fulfillment_status
+    
+    raise ArgumentError, "Fulfillment status not valid" unless VALID_STATUS.include?(@fulfillment_status)
```

Comment:
It's helpful to include the invalid status in the error.

```suggestion
    raise ArgumentError, "Fulfillment status not valid: #{@fulfillment_status}" unless VALID_STATUS.include?(@fulfillment_status)
```